src/fits_schema/binary_table.py

```diff
@@ -165,51 +165,69 @@ def validate_data(self, data, onerror="raise"):
         try:
             # casting = 'safe' makes sure we don't change values
             # e.g. casting doubles to integers will no longer work
-            data = np.asanyarray(data).astype(self.dtype, casting="safe")
+
+            if np.dtype(data.dtype).char in ["S", "U"]:
```

Comment:
I'd merge this for now, but this looks like we should maybe move the actual validation logic to `Column`, not here in `BinTable` using the information of column. So that `Column` subclasses can implement custom validation logic.

Reply:
Yes, I think that will be a separate PR,  since it is a larger re-factor and not really part of just adding support for strings.   Of course that refactor will clean a lot of this up.

Reply:
Note that that check already is inside `Column`, not `BinTable`.  `data` here is the column only, not the full table

Reply:
Right, what I meant was to add this custom logic in the subclasses though, not in the parent.

Reply:
Yes, fully agree, but then I think that is a larger refactoring, as it will change a lot of logic. Here I just wanted to get it to work with minimal changes.  I'll open another PR for #12, and there we can put it correctly in the sub-classes.